clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -598,10 +598,18 @@ public KafkaConsumer(Map<String, Object> configs) {
      * @param valueDeserializer The deserializer for value that implements {@link Deserializer}. The configure() method
      *            won't be called in the consumer when the deserializer is passed in directly.
      */
-    public KafkaConsumer(Map<String, Object> configs,
+    public KafkaConsumer(final Map<String, Object> configs,
                          Deserializer<K> keyDeserializer,
                          Deserializer<V> valueDeserializer) {
-        this(new ConsumerConfig(ConsumerConfig.addDeserializerToConfig(configs, keyDeserializer, valueDeserializer)),
+        this(new ConsumerConfig(
+            ConsumerConfig.addDeserializerToConfig(
+                new Properties() {
```

Comment:
I understand the need for the DBI here on `Properties` and below, but I'm wondering if we could add a static method to `org.apache.kafka.common.utils.CollectionUtils` that takes a map and returns `Properties` instead? (unless that would require an update to the KIP)

Reply:
Seems to be public API... :(